experimental/compactdecimal/src/compactdecimal.rs

```diff
@@ -0,0 +1,235 @@
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    /// <div class="stab unstable">
+    /// ⚠️ The bounds on this function may change over time, including in SemVer minor releases.
+    /// </div>
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_compactdecimal::CompactDecimalFormatter;
+    /// use icu_decimal::options::FixedDecimalFormatterOptions;
+    /// use icu_locid::locale;
+    ///
+    /// CompactDecimalFormatter::try_new_long_unstable(
+    ///     &icu_testdata::unstable(),
+    ///     &locale!("sv").into(),
+    ///     FixedDecimalFormatterOptions::default());
+    /// ```
+    ///
+    /// [data provider]: icu_provider
+    pub fn try_new_long_unstable<D>(
+        data_provider: &D,
+        locale: &DataLocale,
+        options: FixedDecimalFormatterOptions,
+    ) -> Result<Self, CompactDecimalError>
+    where
+        D: DataProvider<LongCompactDecimalFormatDataV1Marker>
+            + DataProvider<icu_decimal::provider::DecimalSymbolsV1Marker>
+            + DataProvider<icu_plurals::provider::CardinalV1Marker>
+            + ?Sized,
+    {
+        Ok(Self {
+            fixed_decimal_format: FixedDecimalFormatter::try_new_unstable(
+                data_provider,
+                locale,
+                options,
+            )?,
+            plural_rules: PluralRules::try_new_cardinal_unstable(data_provider, locale)?,
+            compact_data: DataProvider::<LongCompactDecimalFormatDataV1Marker>::load(
+                data_provider,
+                DataRequest {
+                    locale,
+                    metadata: Default::default(),
+                },
+            )?
+            .take_payload()?
+            .cast(),
+        })
+    }
+
+    /// TODO(egg): Explain that this is an advanced API which can fail (and you
+    /// should use the API that does not exist instead).
+    ///
+    /// # Examples
+    /// ```
+    /// use fixed_decimal::CompactDecimal;
+    /// use icu_compactdecimal::CompactDecimalFormatter;
+    /// use icu_decimal::options::FixedDecimalFormatterOptions;
+    /// use icu_locid::locale;
+    /// use writeable::assert_writeable_eq;
+    /// use std::str::FromStr;
+    ///
+    /// let long_bangla = CompactDecimalFormatter::try_new_long_unstable(
+    ///     &icu_testdata::unstable(),
+    ///     &locale!("bn").into(),
+    ///     FixedDecimalFormatterOptions::default(),
+    /// ).unwrap();
+    ///
+    /// let short_french = CompactDecimalFormatter::try_new_short_unstable(
+    ///     &icu_testdata::unstable(),
+    ///     &locale!("fr").into(),
+    ///     FixedDecimalFormatterOptions::default(),
+    /// ).unwrap();
+    ///
+    /// let long_french = CompactDecimalFormatter::try_new_long_unstable(
+    ///     &icu_testdata::unstable(),
+    ///     &locale!("fr").into(),
+    ///     FixedDecimalFormatterOptions::default(),
+    /// ).unwrap();
+    ///
+    /// let about_a_million = CompactDecimal::from_str("1.2c6").unwrap();
```

Comment:
Suggestion: include a docs test that doesn't render the number so that users are less likely to assume the presence of a number in the output

Reply:
I don’t understand what you mean.

Reply:
One that shows what happens when you format 1c3 in French

Reply:
Ah, yes, good point.